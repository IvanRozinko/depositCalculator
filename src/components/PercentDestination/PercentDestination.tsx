import React from 'react'
import { Typography, Box, Button } from '@material-ui/core';
import { Theme, makeStyles, createStyles } from '@material-ui/core/styles';
import Cash from '@material-ui/icons/AttachMoneySharp';
import Card from '@material-ui/icons/CreditCardSharp';
import { DEPOSIT } from '../../config/constants';

const useStyle = makeStyles((theme: Theme) => 
  createStyles({
    label: {
      width: 140
    },
    buttonsContainer: {
      padding: '0 5%',
      marginRight: 100,
      [theme.breakpoints.down('md')]: {
        flexDirection: 'column'
      }
    },
    helperText: {
      fontSize: 10,
      textAlign: 'center'
    }
  })
);

interface Props {
  destination: string,
  handlePercentDestination: (newDestination: string) => void
}

function PercentDestination({ destination, handlePercentDestination }: Props) {
  const styles = useStyle();
  const isDeposit = destination === DEPOSIT;
  return (
    <Box
      display='flex'
      flexDirection='row'
      alignItems='center'
    >
      <Typography className={styles.label}>
        Проценти
      </Typography>
      <Box
        className={styles.buttonsContainer}
        display='flex'
        flexGrow={1}
        justifyContent='space-between'
      >
        <Box
          display='flex'
          flexDirection='column'
        >
          <Button 
            component='button'
            variant= 'outlined'
            color={isDeposit ? 'primary' : 'default'}
            onClick={() => handlePercentDestination(DEPOSIT)}
            startIcon={<Cash />}
          >
            До вкладу
          </Button>
          <Typography
            className={styles.helperText}
          >
            Збільшується ефективна дохідність
          </Typography>
        </Box>
        <Box
          display='flex'
          flexDirection='column'
        >
        <Button
          component='button'
          variant='outlined'
          color={!isDeposit ? 'primary' : 'default'}
          onClick={() => handlePercentDestination('card')}
          startIcon={<Card />}
        >
          На картку
        </Button>
        <Typography
          className={styles.helperText}
        >
          Проценти можливо відразу витрачати
        </Typography>
        </Box>
      </Box>
    </Box>
  )
}

export default PercentDestination
