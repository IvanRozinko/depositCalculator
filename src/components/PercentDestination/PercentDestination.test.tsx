import PercentDestination from "./PercentDestination";
import { shallow } from 'enzyme';
import { Button } from '@material-ui/core';

describe('<PercentDestination />:', () => {
  const destination = 'card';
  const handlePercentDestination = jest.fn();

  const wrapper = shallow(
    <PercentDestination 
      destination={destination}
      handlePercentDestination={handlePercentDestination}
    />
  )

  it('should renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  })

  it('should call handlePercentDestination on click with certain parameter', () => {
    wrapper.find(Button).at(0).simulate('click');
    expect(handlePercentDestination).toBeCalledWith('deposit');
    wrapper.find(Button).at(1).simulate('click');
    expect(handlePercentDestination).toBeCalledWith('card');
  })
})
