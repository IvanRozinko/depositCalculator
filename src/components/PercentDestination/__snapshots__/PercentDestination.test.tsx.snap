// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PercentDestination />  should renders correctly 1`] = `
<Styled(MuiBox)
  className="makeStyles-container-1"
>
  <WithStyles(ForwardRef(Typography))
    className="makeStyles-label-2"
  >
    Проценти
  </WithStyles(ForwardRef(Typography))>
  <Styled(MuiBox)
    className="makeStyles-buttonsContainer-3"
    display="flex"
    flexGrow={1}
    justifyContent="space-between"
  >
    <Styled(MuiBox)
      display="flex"
      flexDirection="column"
    >
      <WithStyles(ForwardRef(Button))
        color="default"
        component="button"
        onClick={[Function]}
        startIcon={<Memo />}
        variant="outlined"
      >
        До вкладу
      </WithStyles(ForwardRef(Button))>
      <WithStyles(ForwardRef(Typography))
        className="makeStyles-helperText-4"
      >
        Збільшується ефективна дохідність
      </WithStyles(ForwardRef(Typography))>
    </Styled(MuiBox)>
    <Styled(MuiBox)
      display="flex"
      flexDirection="column"
    >
      <WithStyles(ForwardRef(Button))
        color="primary"
        component="button"
        onClick={[Function]}
        startIcon={<Memo />}
        variant="outlined"
      >
        На картку
      </WithStyles(ForwardRef(Button))>
      <WithStyles(ForwardRef(Typography))
        className="makeStyles-helperText-4"
      >
        Проценти можливо відразу витрачати
      </WithStyles(ForwardRef(Typography))>
    </Styled(MuiBox)>
  </Styled(MuiBox)>
</Styled(MuiBox)>
`;

exports[`<PercentDestination />: should renders correctly 1`] = `
<Styled(MuiBox)
  className="makeStyles-container-1"
>
  <WithStyles(ForwardRef(Typography))
    className="makeStyles-label-2"
  >
    Проценти
  </WithStyles(ForwardRef(Typography))>
  <Styled(MuiBox)
    className="makeStyles-buttonsContainer-3"
    display="flex"
    flexGrow={1}
    justifyContent="space-between"
  >
    <Styled(MuiBox)
      display="flex"
      flexDirection="column"
    >
      <WithStyles(ForwardRef(Button))
        color="default"
        component="button"
        onClick={[Function]}
        startIcon={<Memo />}
        variant="outlined"
      >
        До вкладу
      </WithStyles(ForwardRef(Button))>
      <WithStyles(ForwardRef(Typography))
        className="makeStyles-helperText-4"
      >
        Збільшується ефективна дохідність
      </WithStyles(ForwardRef(Typography))>
    </Styled(MuiBox)>
    <Styled(MuiBox)
      display="flex"
      flexDirection="column"
    >
      <WithStyles(ForwardRef(Button))
        color="primary"
        component="button"
        onClick={[Function]}
        startIcon={<Memo />}
        variant="outlined"
      >
        На картку
      </WithStyles(ForwardRef(Button))>
      <WithStyles(ForwardRef(Typography))
        className="makeStyles-helperText-4"
      >
        Проценти можливо відразу витрачати
      </WithStyles(ForwardRef(Typography))>
    </Styled(MuiBox)>
  </Styled(MuiBox)>
</Styled(MuiBox)>
`;
